src/Agent/NewRelic/Agent/Core/Configuration/AppSettingsConfigResolveWhenUsed.cs

```diff
@@ -8,55 +8,68 @@
 
 namespace NewRelic.Agent.Core.Configuration
 {
-	public static class AppSettingsConfigResolveWhenUsed
-	{
-		private static IConfiguration _configuration;
-		private static string _appSettingsFilePaths;
-
-		private static IConfiguration Configuration
-		{
-			get
-			{
-				return _configuration ?? (_configuration = InitializeConfiguration());
-			}
-		}
-
-		private static IConfigurationRoot InitializeConfiguration()
-		{
-			var env = new SystemInterfaces.Environment();
-			var currentDirectory = Directory.GetCurrentDirectory();
-			var environment = env.GetEnvironmentVariable("EnvironmentName");
-
-			var builder = new ConfigurationBuilder()
-				.SetBasePath(currentDirectory)
-				.AddJsonFile("appsettings.json", optional: true, reloadOnChange: false);
-
-			if (!string.IsNullOrEmpty(environment)) builder.AddJsonFile($"appsettings.{environment}.json", optional: true, reloadOnChange: false);
-
-			var appSettingsPath = Path.Combine(currentDirectory, "appsettings.json");
-			var appSettingsEnvPath = Path.Combine(currentDirectory, $"appsettings.{environment}.json");
-			_appSettingsFilePaths = !string.IsNullOrEmpty(environment) ? string.Join(", ", appSettingsPath, appSettingsEnvPath) : appSettingsPath;
-
-			return builder.Build();
-		}
-
-		public static string GetAppSetting(string key)
-		{
-			if (key == null)
-			{
-				return null;
-			}
-			
-			var value = Configuration[key];
-
-			if (Log.IsDebugEnabled)
-			{
-				if (string.IsNullOrWhiteSpace(value)) Log.Debug($"Reading value from appsettings.json: '{key}' not defined. Searched: {_appSettingsFilePaths}.");
-				else Log.Debug($"Reading value from appsettings.json: '{key}={value}'");
-			}
-
-			return value;
-		}
-	}
+    public static class AppSettingsConfigResolveWhenUsed
+    {
+        private static IConfiguration _configuration;
+        private static string _appSettingsFilePaths;
+
+        private static IConfiguration Configuration
+        {
+            get
+            {
+                return _configuration ?? (_configuration = InitializeConfiguration());
+            }
+        }
+
+        private static IConfigurationRoot InitializeConfiguration()
+        {
+            var env = new SystemInterfaces.Environment();
```

Comment:
Slight perf improvement: Looks like this var declaration can be moved inside the if on line 29, since that is the only place it is used and may be avoided if ASPNETCORE_ENVIRONMENT is defined.